frontend/src/project/classes/Project.ts

```diff
@@ -0,0 +1,117 @@
+import { IPairingBoard } from "../interfaces/IPairingBoard";
+import { IPerson } from "../interfaces/IPerson";
+
+export interface IProject {
+  name: string;
+  id: number;
+  pairingBoards: IPairingBoard[];
+  people: IPerson[];
+}
+
+export class Project implements IProject {
+  name: string;
+  id: number;
+  pairingBoards: IPairingBoard[];
+  people: IPerson[];
+
+  constructor(project: IProject) {
+    this.name = `${project.name}`;
+    this.id = project.id;
+    this.pairingBoards = [...project.pairingBoards];
+    this.people = [...project.people];
+  }
+
+  get canAPairingBeMade(): boolean {
+    const atLeast2Floaters = this.people.length >= 2;
+    const atLeast1Floater = this.people.length >= 1;
+    const atLeast1EmptyBoard = this.emptyPairingBoard !== undefined;
+    const atLeast1UnpairedSticker =
+      this.currentUnpairedStickingPeople.length >= 1;
+    return (
+      (atLeast1Floater && atLeast1UnpairedSticker) ||
+      (atLeast2Floaters && atLeast1EmptyBoard)
+    );
+  }
+
+  get emptyPairingBoard() {
+    return this.pairingBoards.find((pb) => pb.people.length === 0);
```

Comment:
```suggestion
    return this.pairingBoards.some((pb) => pb.people.length === 0);
```
These two behave the same way but I think ```some``` here might be more clear since it returns a boolean.

Reply:
no, this is being used to retrieve the actual board in at least one place. If it were being a boolean I would agree. I also didn't know about `.some` and might have constructed this differently had I known.